py-polars/polars/lazyframe/frame.py

```diff
@@ -2602,11 +2634,60 @@ def filter(self, predicate: IntoExpr) -> Self:
         └─────┴─────┴─────┘
 
         """
-        if isinstance(predicate, list):
-            predicate = pl.Series(predicate)
+        # note: identify masks separately from predicates
+        all_predicates, boolean_masks = [], []
+        for p in predicates:
+            if is_bool_sequence(p):
+                boolean_masks.append(pl.Series(p, dtype=Boolean))
+            else:
+                all_predicates.append(parse_as_expression(p, wrap=True))
+
+        # identify deprecated usage of 'predicate' parameter
+        if "predicate" in constraints:
+            is_mask = False
+            if isinstance(p := constraints["predicate"], pl.Expr) or (
+                is_mask := is_bool_sequence(p)
+            ):
+                p = constraints.pop("predicate")
+                warnings.warn(
+                    "`filter` no longer takes a 'predicate' parameter.\n"
+                    "To silence this warning you should omit the keyword and pass "
+                    "as a positional argument instead.",
+                    DeprecationWarning,
+                    stacklevel=find_stacklevel(),
+                )
+                if is_mask:
+                    boolean_masks.append(pl.Series(p, dtype=Boolean))
+                else:
+                    all_predicates += (p,)
```

Comment:
Just use `append` here, like on line 2643?
```suggestion
                    all_predicates.append(p)
```

Reply:
Ahh! Good spot - in an earlier draft I was appending onto the 'predicates' tuple where yo can't append 😅